api/cpp/cmake/SlintConfig.cmake.in

```diff
@@ -14,8 +14,41 @@ endif()
 add_library(@slint_cpp_impl@ @cmake_lib_type@ IMPORTED)
 set_target_properties(@slint_cpp_impl@ PROPERTIES @SLINT_LIB_PROPERTIES@)
 
-set(SLINT_COMPILER @SLINT_COMPILER@ CACHE FILEPATH "Path to the slint-compiler executable")
+function(_slint_download_compiler_and_cache)
+    set(SLINT_GITHUB_RELEASE @SLINT_VERSION@ CACHE STRING "GitHub Release to use for Slint Binary Artifact Downloads")
+
+    if (CMAKE_HOST_WIN32)
+        set(compiler_exe_suffix ".exe")
+        set(platform_suffix "windows")
+    else()
+        set(platform_suffix "${CMAKE_HOST_SYSTEM_NAME}-${CMAKE_HOST_SYSTEM_PROCESSOR}")
+    endif()
+
+    set(prebuilt_archive_filename "slint-compiler-${platform_suffix}.tar.gz")
+    set(download_url "https://github.com/slint-ui/slint/releases/download/${SLINT_GITHUB_RELEASE}/${prebuilt_archive_filename}")
+    set(download_directory "${CMAKE_BINARY_DIR}/slint-prebuilt")
+    message(STATUS "Downloading pre-built Slint compiler binary from ${download_url}")
+    file(DOWNLOAD "${download_url}" "${download_directory}/${prebuilt_archive_filename}" STATUS download_status)
```

Comment:
did you not say you want to check if the file exist before downloading?

Reply:
I realized that's not necessarily because the existence of the SLINT_COMPILER cache entry as != download suffices to enable quick re-configuration runs.